crates/snforge-scarb-plugin/src/attributes/should_panic/expected.rs

```diff
@@ -0,0 +1,79 @@
+use super::ShouldPanicCollector;
+use crate::{
+    attributes::{AttributeInfo, ErrorExt},
+    cairo_expression::CairoExpression,
+    types::{Felt, ParseFromExpr},
+};
+use cairo_lang_macro::Diagnostic;
+use cairo_lang_syntax::node::{ast::Expr, db::SyntaxGroup};
+
+#[derive(Debug, Clone, Default)]
+pub enum Expected {
+    Felt(Felt),
+    ByteArray(String),
+    Array(Vec<Felt>),
+    #[default]
+    Any,
+}
+
+impl CairoExpression for Expected {
+    fn as_cairo_expression(&self) -> String {
+        match self {
+            Self::Felt(felt) => {
+                let string = felt.as_cairo_expression();
+
+                format!("snforge_std::_config_types::Expected::ShortString({string})")
+            }
+            Self::ByteArray(string) => {
+                let string = string.as_cairo_expression();
+
+                format!(r#"snforge_std::_config_types::Expected::ByteArray({string})"#)
+            }
+            Self::Array(strings) => {
+                let arr = strings.as_cairo_expression();
+
+                format!("snforge_std::_config_types::Expected::Array({arr})")
+            }
+            Self::Any => "snforge_std::_config_types::Expected::Any".to_string(),
+        }
+    }
+}
+
+impl ParseFromExpr<Expr> for Expected {
+    fn parse_from_expr<T: AttributeInfo>(
+        db: &dyn SyntaxGroup,
+        expr: &Expr,
+        arg_name: &str,
+    ) -> Result<Self, Diagnostic> {
+        let error_msg = format!(
+            "<{arg_name}> argument must be string, short string, number or list of short strings or numbers in regular brackets ()"
+        );
+
+        match expr {
+            Expr::ShortString(_) | Expr::Literal(_) => {
+                Ok(Self::Felt(
+                    Felt::parse_from_expr::<ShouldPanicCollector>(db, expr, arg_name)
+                        // this unwrap is safe because we checked if expression is valid short string or number
+                        .unwrap(),
+                ))
```

Comment:
In the match

```rust
match expr {
            Expr::ShortString(_) | Expr::Literal(_) => { <--- here
                Ok(Self::Felt(
                    Felt::parse_from_expr::<ShouldPanicCollector>(db, expr, arg_name)
                        // this unwrap is safe because we checked if expression is valid short string or number
                        .unwrap(),
                ))
            }
```